test/e2e/pod.go

```diff
@@ -94,34 +84,32 @@ var _ = ginkgo.Describe("Pod to external server PMTUD", func() {
 			}
 			e2epod.NewPodClient(f).CreateSync(context.TODO(), clientPod)
 
-			ginkgo.By("Creating the server pod with hostNetwork:true")
-			// Create the server pod.
-			// Wait for 1 minute and if the pod does not come up, select a different port and try again.
-			// Wait for a max of 5 minutes.
-			gomega.Eventually(func() error {
-				serverPodPort = rand.Intn(echoServerPodPortMax-echoServerPodPortMin) + echoServerPodPortMin
-				serverPodName = fmt.Sprintf(echoServerPodNameTemplate, serverPodPort)
-				framework.Logf("Creating server pod listening on TCP and UDP port %d", serverPodPort)
-				serverPod = e2epod.NewAgnhostPod(f.Namespace.Name, serverPodName, nil, nil, nil, "netexec",
-					"--http-port",
-					fmt.Sprintf("%d", serverPodPort),
-					"--udp-port",
-					fmt.Sprintf("%d", serverPodPort))
-				serverPod.ObjectMeta.Labels = map[string]string{
-					"app": serverPodName,
-				}
-				serverPod.Spec.HostNetwork = true
-				serverPod.Spec.NodeName = serverPodNodeName
-				e2epod.NewPodClient(f).Create(context.TODO(), serverPod)
-
-				err := e2epod.WaitTimeoutForPodReadyInNamespace(context.TODO(), f.ClientSet, serverPod.Name, f.Namespace.Name, 1*time.Minute)
-				if err != nil {
-					e2epod.NewPodClient(f).Delete(context.TODO(), serverPod.Name, metav1.DeleteOptions{})
-					return err
-				}
-				serverPod, err = e2epod.NewPodClient(f).Get(context.TODO(), serverPod.Name, metav1.GetOptions{})
-				return err
-			}, 5*time.Minute, 1*time.Second).Should(gomega.Succeed())
+			ginkgo.By("Creating the external server")
+			serverPodPort = rand.Intn(echoServerPodPortMax-echoServerPodPortMin) + echoServerPodPortMin
+			serverPodName = fmt.Sprintf(echoServerPodNameTemplate, serverPodPort)
+			framework.Logf("Creating server pod listening on TCP and UDP port %d", serverPodPort)
+			agntHostCmds := []string{"netexec", "--http-port", fmt.Sprintf("%d", serverPodPort), "--udp-port", fmt.Sprintf("%d", serverPodPort)}
+			externalIpv4, externalIpv6 := createClusterExternalContainer(serverPodName, agnhostImage,
+				[]string{"--network", "kind", "-P", "--cap-add", "NET_ADMIN"},
+				agntHostCmds,
+			)
+
+			if isIPv4Supported() {
+				serverNodeInternalIPs = append(serverNodeInternalIPs, externalIpv4)
+			}
+
+			if isIPv6Supported() {
+				serverNodeInternalIPs = append(serverNodeInternalIPs, externalIpv6)
+			}
+
+			gomega.Expect(len(serverNodeInternalIPs)).To(gomega.BeNumerically(">", 0))
+		})
+
+		ginkgo.AfterEach(func() {
```

Comment:
there is an afterEach at L33 but is that common to whole suite which is why you did this internal aftereach only for this test?

Reply:
It's not common to the whole suite because the next set of tests are run against a host network pod with a different When clause:
		ginkgo.When("tests are run towards the agnhost echo server", func() {
			ginkgo.It("queries to the hostNetworked server pod on another node shall work for TCP", func() {


...but that being said, I think these tests should be duplicated to also use an external server in the PR where I add support for accessing host ip via overlay. This is because this test was testing that the GR would send ICMP needs frag, and once my PR changes behavior to use mp0 we will lose some coverage here. I will address that in https://github.com/ovn-org/ovn-kubernetes/pull/4069

Reply:
oops I read the indentations wrong so deleted my last comment. I guess it could be there, but the beforeEach already existed at this scope so I just preserved the afterEach in the same scope.